### DISCUSSION TITLE
The F12 idea is rather quite interesting!

### DISCUSSION THREAD

**Discussion by gcarreno** [gcarreno/laz-JSON-Viewer#14] [Ideas]
Hey Craig (@chapmanworld)

You actually have a very good idea with the F12.

After I go through the current list of self inflicted issues, I'm definitely gonna try to implement this!

But I will not attempt to synch text and tree like I discussed here: #13 
That is still a bit advanced for my meagre skills, I think!

**Comment by MFernstrom** [comment]
If you use a hotkey to switch between them, you don't need to synch with caret position. You could synch the content when switching view.

**Reply by gcarreno** [reply]
That is a wonderful idea and it would work if, somewhere inside `fpjson`, we can have that connection between `TJSONData` and a line on a String.

I'm afraid the problem is a bit more complicated, both ways: Tracking from `TJSONData` to String and/or from String to `TJSONData`.

Some, very hacky, ideas come to mind, but I'm not even sure even those would work.

Do you have some specific ideas you want to share? I am quite stumped by it and REALLY welcome any outside input.

The only thing I can, quickly, think of that could work, is to do the JSON parsing myself. That completely invalidates the use of `fpjson` which is so easy for the picking.

At the moment the record for the `VST` is keeping a bit more info than I would want to, but it's not that hacky yet.
I'm implementing the display of the path, when you click on a node, with data I have to pump into that record while doing the recursive build of the tree.
That means `fpjson` can't even give me that. Unless I'm missing something very in my face.
I can ask to give me a node/value if I provide a path, but the `TJSONData` does not have _awareness of it's environment_, as it be :disappointed:
